contracts/oracles/SFrxETHOracle.sol

```diff
@@ -1,29 +1,87 @@
+contract SFrxETHOracle is AccessControlledV8 {
+    /// @notice Address of SfrxEthFraxOracle
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    ISfrxEthFraxOracle public immutable SFRXETH_FRAX_ORACLE;
+
+    /// @notice Address of sfrxETH
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable SFRXETH;
+
+    /// @notice Maximum allowed price difference
+    uint256 public maxAllowedPriceDifference;
+
+    /// @notice Emits when the maximum allowed price difference is updated
+    event MaxAllowedPriceDifferenceUpdated(uint256 oldMaxAllowedPriceDifference, uint256 newMaxAllowedPriceDifference);
+
+    /// @notice Thrown if the price data is invalid
+    error BadPriceData();
+
+    /// @notice Thrown if the price difference exceeds the allowed limit
+    error PriceDifferenceExceeded();
+
+    /// @notice Thrown if the token address is invalid
+    error InvalidTokenAddress();
+
     /// @notice Constructor for the implementation contract.
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(
-        address sfrxETH,
-        address frxETH,
-        address resilientOracle
-    ) CorrelatedTokenOracle(sfrxETH, frxETH, resilientOracle) {}
+    constructor(address _sfrxEthFraxOracle, address _sfrxETH) {
+        ensureNonzeroAddress(_sfrxEthFraxOracle);
+        ensureNonzeroAddress(_sfrxETH);
+        SFRXETH_FRAX_ORACLE = ISfrxEthFraxOracle(_sfrxEthFraxOracle);
+        SFRXETH = _sfrxETH;
+    }
 
     /**
-     * @notice Gets the frxETH for 1 sfrxETH
-     * @return amount Amount of frxETH
+     * @notice Sets the contracts required to fetch prices
+     * @param _accessControlManager Address of the access control manager contract
      */
-    function _getUnderlyingAmount() internal view override returns (uint256) {
-        return ISfrxETH(CORRELATED_TOKEN).convertToAssets(EXP_SCALE);
+    function initialize(address _accessControlManager) external initializer {
+        __AccessControlled_init(_accessControlManager);
+    }
+
+    /**
+     * @notice Sets the maximum allowed price difference
+     * @param _maxAllowedPriceDifference Maximum allowed price difference
+     */
+    function setMaxAllowedPriceDifference(uint256 _maxAllowedPriceDifference) external {
+        _checkAccessAllowed("setMaxAllowedPriceDifference(uint256)");
+        emit MaxAllowedPriceDifferenceUpdated(maxAllowedPriceDifference, _maxAllowedPriceDifference);
+        maxAllowedPriceDifference = _maxAllowedPriceDifference;
+    }
+
+    /**
+     * @notice Fetches the USD price of sfrxETH
+     * @param asset Address of the sfrxETH token
+     * @return price The price scaled by 1e18
+     */
+    function getPrice(address asset) external view returns (uint256) {
+        if (asset != SFRXETH) revert InvalidTokenAddress();
+
+        (bool isBadData, uint256 priceLow, uint256 priceHigh) = SFRXETH_FRAX_ORACLE.getPrices();
+
+        if (isBadData) revert BadPriceData();
```

Comment:
For full disclosure we do not revert on this but either should be fine. 

![Screenshot 2024-05-10 at 1 16 01 PM](https://github.com/VenusProtocol/oracle/assets/88335455/bff61b9e-0dfc-4317-9303-8073496537e7)

This will just make warping on fork testing a pain